spec/policies/exercise_policy_spec.rb

```diff
@@ -48,6 +48,14 @@
     end
   end
 
+  [:show?].each do |action|
```

Comment:
you changed the logic. now a teacher can show (see) all exercises. This is not a problem, but I just want to note it here.